windpowerlib/power_output.py

```diff
@@ -238,14 +228,26 @@ def power_curve_density_correction(
             Variable Scale Simulation Model for Windpower based on the
             Georeferenced Installation Register of Germany". Master's Thesis
             at Reiner Lemoine Institute, 2014, p. 13
-
     """
     if density is None:
         raise TypeError(
             "`density` is None. For the calculation with a "
             + "density corrected power curve density at hub "
             + "height is needed."
         )
+
+    # NOTE : CHANGES ARE MADE HERE FOR SPEED IMPROVEMENT
+    # create a flag for pandas Series type
+    Panda_series = False
+
+    if isinstance(wind_speed, pd.Series):
+        # save the indexes for later conversion to pd.Series
+        indexes = wind_speed.index
```

Comment:
Why not `wind_speed_index`?

Reply:
Sorry for the mistakes and thank you for pointing out the suggestions. This does improves the readability of code.
Please check the new commit for changes.